### PR TITLE
[58] [59] [60] [UI+Integration] As a user, I can see the Filter screen

### LINKED ISSUE #58: [UI] As a user, I can see the sort type selection on the Filter screen
## Why

As a user, I can see the sort type selection on the Filter screen
 
## Acceptance criteria

- The sort type selection contains 3 items: `New`, `Price (High to low)` and `Price (Low to high)`
- Each item has:
    - Title label
    - Description label
    - Radio button
- The user can tap on an item to change the selection
- The default selection is `New`

## Resource

image

### LINKED ISSUE #59: [UI] As a user, I can see the filter options section on the Filter screen
## Why

As a user, I can see the sort type selection on the Filter screen
 
## Who Benefits?

- The `Filter by` label
- There are 2 filter options: `Categories` and `Colors`
- Each filter option has:
    - Title label
    - Disclosure image

## Resource

image

### LINKED ISSUE #60: [Integrate] As a user, I can see the Filter screen
## Why

As a user, I can see the Filter screen after tapping on the filter button on the Search Result screen
 
## Acceptance criteria

As a user, I can see the Filter screen after tapping on the filter button on the Search Result screen

The Filter screen contains:

- The navigation bar with:
    - `Close` button on the left corner
    - `Filters` title in the middle
    - `Reset` button on the right corner
- The sort type selection section, which is implemented in the issue https://github.com/nimblehq/nimble-ecommerce-ios/issues/58
- The filter options section, which is implemented in the issue https://github.com/nimblehq/nimble-ecommerce-ios/issues/59
- `View results` button at the bottom of the screen

## Resource

image

### PR DESCRIPTION
**Pull request by nmint8m**
https://github.com/nimblehq/nimble-ecommerce-ios/issues/58
https://github.com/nimblehq/nimble-ecommerce-ios/issues/59
https://github.com/nimblehq/nimble-ecommerce-ios/issues/60

## What happened 👀

As a user, I can see the Filter screen after tapping on the filter button on the Search Result screen

The Filter screen contains:

- The navigation bar with:
    - `Close` button on the left corner
    - `Filters` title in the middle
    - `Reset` button on the right corner
- The sort type selection section, which is implemented in the issue https://github.com/nimblehq/nimble-ecommerce-ios/issues/58
- The filter options section, which is implemented in the issue https://github.com/nimblehq/nimble-ecommerce-ios/issues/59
- `View results` button at the bottom of the screen

### CODE REVIEW

**Comment by vnntsu** [comment]
@nmint8m I don't see `View results` button at the bottom

File: ecommerce-ios/Sources/Filter/FilterScreen.swift [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                titleText("Filter by")
+
+                ForEach(filterViewModels, id: \.self) { filterViewModel in
+                    FilterCell(viewModel: filterViewModel.rawValue)
+                }
+            }
+        }
+        .navigationTitle("Filters")
+        .navigationBarItems(
+            leading: Button(action: { presentationMode.wrappedValue.dismiss() } ) { Text("Close") }
```
Comment (blyscuit) [inline ecommerce-ios/Sources/Filter/FilterScreen.swift]: should we break this, currently it's dense.

File: ecommerce-ios/Views/SortSelection/SortSelectionView.swift [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    
+    var body: some View {
+        VStack(spacing: 0.0) {
+            ForEach(cellViewModels) { cellViewModel in
+                ZStack(alignment: .trailing) {
+                    RadioButtonView(viewModel: selectedViewModel == cellViewModel)
+                        .padding(.horizontal, 16.0)
+
+                    SortCell(viewModel: cellViewModel)
+                        .frame(height: 44)
```
Comment (blyscuit) [inline ecommerce-ios/Views/SortSelection/SortSelectionView.swift]:
```suggestion
                        .frame(height: 44.0)
```

**Comment by nmint8m** [comment]
> @nmint8m I don't see `View results` button at the bottom

@vnntsu I just added the button. Please check again.

File: ecommerce-ios/Views/ColorCell/ColorCell.swift [hunk 3]
```diff
@@ -23,6 +23,7 @@ struct ColorCell: View {
                         .font(.system(size: 13.0))
                 }
             }
+            .contentShape(Rectangle())
```
Comment (blyscuit) [inline ecommerce-ios/Views/ColorCell/ColorCell.swift:26]: I would say another way to do this view is
```
var body: some View {
VStack(spacing: 8.0) {
            Circle()
                .foregroundColor(Color(hexString: viewModel.colorCode))
                .frame(width: 24.0, height: 24.0)
            Text(viewModel.name.capitalized)
                .foregroundColor(.charadeGray)
                .font(.system(size: 13.0))
        }
.frame(maxWidth: .infinity, idealHeight: 72.0)
}
```

But I'm not sure what's the benefit compares to .overlay

**Review by blyscuit — APPROVED** [review]

File: ecommerce-ios/Views/ColorCell/ColorCell.swift (hunk 3 above)
Comment (nmint8m) [reply ecommerce-ios/Views/ColorCell/ColorCell.swift:26]: It is my workaround solution. 

So without `.contentShape(Rectangle())`, the `ColorView` is only tappable when I tap on the circle and text, the spacing around is not tappable.

Basically, the `.contentShape(Rectangle())` makes the whole view tappable, and it resolves my problem 😅

I will give your solution a try and let you know if it works on swiftui channel 👍

**Review by vnntsu — APPROVED** [review]
LGTM!

**Pull request merged**